### ISSUE TITLE
Reporting emphasis <em> and strong <strong> in browsers

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#17216]
### Steps to reproduce:

Reported by Ivan via email.

1. Open a page with text with emphasis.  On the NVDA Download page at https://www.nvaccess.org/download/ the text "Please note: NVDA is only available for PCs running Microsoft Windows 8.1 and later." uses <strong> and the text "NV Access does not recommend or support running these older versions on newer Operating systems." uses <em>
2. On NVDA's document formatting settings (NVDA+control+d) ensure "emphasis" is checked.
3. Read the above mentioned text.

### Actual behavior:

NVDA reads the text without noting any emphasis.  If font attributes are enabled in NVDA document formatting settings, then the <strong> text is reported as bold and <em> text as italic.

### Expected behavior:

Text using <strong> should be reported as "Strong" and text using <em> should be reported as "Emphasised".

[This comment on a previous issue](https://github.com/nvaccess/nvda/issues/4920#issuecomment-161162498) seems to indicate that this behaviour used to work.

#### NVDA version:

NVDA 2024.4 Beta 4 installed

#### Windows version:

Windows 11 (64-bit) Version: 23H2, Build: 22631.4169 

#### Name and version of other software in use when reproducing the issue:

Firefox version: 130.0.1 
 
Chrome Version: 129.0.6668.59 

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

I tried NVDA 2018.4.1 (released a few months after that linked comment) but with the current browser versions, and the text was read identically to how it is read with NVDA 2024.4 beta 4.

#### If NVDA add-ons are disabled, is your problem still occurring?

No add-ons.

**Comment by CyrilleB79** [comment]
It's worth noting that in MSHTML `<em>` and `<strong>` are reported twice, once if "Attributes" is checked and once if Emphasis is checked in Document formatting settings.

MSHTML was used with Internet Explorer, but it is still used in `ui.browseableMessage`. So you can test in Python console:
```
import ui;ui.browseableMessage('<p>This is an example for<strong>strongified</strong>or<em>emphasized</em>text</p>', isHtml=True)
```

Moreover `<strong>` is reported as "bold" because it has been argued in the past than "strong" would not be understood by many users.

**Comment by Adriani90** [comment]
I agree reporting "strong" is not common. When communicating with sighted people, the word "bold" or "emphasized" is rather used for text that stands out.

**Comment by ivnc** [comment]
I think that despite bold is visually correct for labelling this kind of text, we should still distinguish between actual bold (b tag) and emphasis (strong or em), both when reporting it and also in the document formatting options, as it is currently. The problem is that the emphasis option seems not to work, and that prevents users from A. Being able to enable b and strong/em reporting separately, and B. Identifying whether a text is marked as bold only for visual purposes or as strong/em because it is actually highlighted in the page, something between a simple bold and a heading.

**Comment by seanbudd** [comment]
This can't be fixed due to issues on the browser side. An issue needs to be opened on the chromium and firefox bug trackers. In the meantime, we should remove this option from NVDA.

**Comment by CyrilleB79** [comment]
The option is working in MSHTML, which is used for browseable messages, and maybe in other components.

Maybe it does not hurt to keep the option?

**Comment by seanbudd** [comment]
I don't think MSHTML usage really justifies the feature, it creates confusion for users. I do think removing it would be an API breaking change.

**Comment by seanbudd** [comment]
I'm not sure if this information is exposed via the IAccessible buffer in Firefox, it's possible that nothing needs to be done if they are exposed in the accessibility tree, but the tree doesn't always match whats exposed to the screen reader. 

@michaelDCurran - are we sure this is still blocked on the browser side. These bugs that seem solved in VoiceOver make me think this work is unblocked:
- https://bugzilla.mozilla.org/show_bug.cgi?id=1740610
- https://issues.chromium.org/issues/40694879
- https://issues.chromium.org/issues/40671440

**Comment by jcsteh** [comment]
Firefox and Chrome both expose objects in the tree for `<em>` and `<strong>`. However:

1. The spec says that these should map to the ARIA roles emphasis and strong.
2. This in turn means browsers should expose object attribute xml-roles:emphasis and xml-roles:strong. Neither engine does.
3. Both engines expose tag:em and tag:strong, so clients could use that. However, that would mean you'd need to handle the tags and the ARIA roles separately, which is not ideal, though NVDA is already doing this for `<figure>` in Chrome.

The browser bug here is that `<em>` and `<strong>` should expose xml-roles:emphasis and xml-roles:strong, respectively. I filed a Firefox bug for this: https://bugzilla.mozilla.org/show_bug.cgi?id=1938268

**Comment by ivnc** [comment]
@jcsteh thank you so much for the time and the explanation. If I am not wrong, in the Bugzilla issue description you say that setting the ARIA role explicitly would be enough to have it exposed in the right way to screen readers. However even setting roles emphasis or strong explicitly I can't have that content read by NVDA, could there be, apart from the browsers' issue, some kind of incomplete implementation within NVDA or so? No idea at all about this so technical topics, but the fact is that NVDA doesn't read nor the tags nor the explicitly set roles.

**Comment by jcsteh** [comment]
You're correct: work is needed on the NVDA side. I was only answering the question of what browsers need to do here, not what NVDA needs to do.

NVDA needs to map xml-roles:emphasis and xml-roles:strong correctly. Once it does that, role="emphasis" and role="strong" will work in both browsers. However, even once that's done, `<em>` and `<strong>` without an ARIA role won't work until the browser bugs are fixed.

I will flag that we turned on reading of emphasis by default a few years ago with MSHTML and users absolutely hated it. While `<em>` and `<strong>` theoretically provide better semantics than `<b>` and `<i>`, they are not used wisely in practice.
